samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -281,83 +289,115 @@ static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup
+		       data.bis[i].chan_allocation);
 	}
 
 	/* Get the BIS indexes */
 	for (uint8_t i = 0U; i < data.cnt; i++) {
 		if ((data.bis[i].chan_allocation == CONFIG_TARGET_BROADCAST_CHANNEL) ||
-		    ((data.bis[i].chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) ==
-		     CONFIG_TARGET_BROADCAST_CHANNEL)) {
-			/* Exact match */
+		    (CONFIG_TARGET_BROADCAST_CHANNEL == BT_AUDIO_LOCATION_MONO_AUDIO)) {
+			/* Exact match (accept all allocations as mono) */
 			bis_indexes = BT_ISO_BIS_INDEX_BIT(data.bis[i].index);
 
-			printk("Channel allocation match. BIS index bitfield 0x%x\n", bis_indexes);
-			*(uint32_t *)user_data = bis_indexes;
-
-			return false; /* bis index found */
+			printk("Channel allocation match, exact. BIS idx bitfield 0x%x\n",
+			       bis_indexes);
+			bis_index->bis_index_bitfield[bis_index->num_subgroups] = bis_indexes;
+			break; /* no need to go futher */
 		} else if ((data.bis[i].chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) != 0) {
-			/* Partial match */
 			chan_allocation |= data.bis[i].chan_allocation;
 			bis_indexes |= BT_ISO_BIS_INDEX_BIT(data.bis[i].index);
 
 			if ((chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) ==
 			    CONFIG_TARGET_BROADCAST_CHANNEL) {
-				printk("Channel allocation match. BIS index bitfield 0x%x\n",
+				/* Exact match */
+				printk("Channel allocation match, exact. BIS idx bitfield 0x%x\n",
 				       bis_indexes);
-				*(uint32_t *)user_data = bis_indexes;
 
-				return false; /* bis indexes found */
+				bis_index->bis_index_bitfield[bis_index->num_subgroups] =
+					bis_indexes;
+				break; /* no need to go futher */
 			}
+
+			/* Partial match */
+			printk("Channel allocation match, partial. BIS idx bitfield 0x%x\n",
+			       bis_indexes);
 		}
 	}
+
+	bis_index->num_subgroups++;
+
 	return true; /* continue to next subgroup */
 }
+#else /* !CONFIG_TARGET_BROADCAST_CHANNEL */
+static bool subgroup_get_bis_indexes_cb(const struct bt_bap_base_subgroup *subgroup,
+					void *user_data)
+{
+	struct bis_index_data *bis_index = user_data;
+	const int err = bt_bap_base_subgroup_get_bis_indexes(
+		subgroup, &bis_index->bis_index_bitfield[bis_index->num_subgroups]);
+
+	if (err != 0) {
+		printk("Failed to parse all BIS: %d", err);
+
+		return false;
+	}
+
+	printk("bis_index.bis_index_bitfield[%d] = 0x%0x\n", bis_index->num_subgroups,
+	       bis_index->bis_index_bitfield[bis_index->num_subgroups]);
+
+	bis_index->num_subgroups++;
+
+	return true;
+}
 #endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
 
 static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap_base *base,
 			 size_t base_size)
 {
-	uint32_t base_bis_index_bitfield = 0U;
 	int err;
 
+	base_bis_index_bitfield = 0U;
+
 	if (base_received) {
 		return;
 	}
 
 	printk("Received BASE with %d subgroups from broadcast sink %p\n",
 	       bt_bap_base_get_subgroup_count(base), sink);
 
+	memset(&bis_index, 0, sizeof(struct bis_index_data));
```

Comment:
```suggestion
	(void)memset(&bis_index, 0, sizeof(struct bis_index_data));
```

Our most recent coding guidelines require either handling or explicitly ignoring return values (rule 100 of https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html)